### PR TITLE
feat(trading): data grid dropdown menu in collateral table and vega icons

### LINKED ISSUE #3213: Datagrid row actions
## Story

As a user
I want more space in datagrids
So that I can see more information at any given time and not have repeated button text

## Acceptance Criteria

- [ ] I can perform already existing row actions
  - [ ] Order list: edit, cancel 
  - [ ] Positions: close
  - [ ] Collateral: deposit, withdraw
- [ ] Datagrids with multiple actions are combined into a single UI pattern

## Tasks

- [x] Explore and sketch
- [x] Visual Design
  - [ ] Create icons for available actions: deposit, withdraw, edit, cancel, close, breakdown (use info icon), transfer
- [ ] Ui toolkit development
  - [ ] Adjust styles to match designs
  - [ ] Allow dropdown placement to be set via props, (row dropdown should render to the left of the kebab button)

## Additional details / background info

Currently in some tables we end up with a lot of repeated button text, it already takes up lots of space and if we continue with this pattern we'll lose even more (what if a table row as 5/6 plus actions?).

Current example:
Screenshot 2023-03-15 at 18 41 49

### PR DESCRIPTION
**Pull request by MadalinaRaicu**
# Related issues 🔗

Closes #3213 

# Description ℹ️

Add data grid dropdown in the collateral table and update overlay on dialog only on dark mode to match design.

# Demo 📺

Screenshot 2023-04-21 at 16 51 03

Screenshot 2023-04-19 at 18 17 50

Screenshot 2023-04-19 at 18 19 31

Screenshot 2023-04-19 at 18 19 25

Screenshot 2023-04-19 at 18 18 25

### CODE REVIEW

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu-grid.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+import classNames from 'classnames';
+import { forwardRef } from 'react';
+import { Icon } from '../icon';
+
+/**
+ * The button that toggles the dropdown menu inside grids.
+ * By default, the {@link DropdownMenuContent} will position itself against the trigger.
+ */
+export const DropdownMenuGridTrigger = forwardRef<
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu-grid.tsx:10]: chore (blocking): The icon that you choose to render is the apps responsibility, we shouldn't add components to handle this, what happens when someone else needs a dropdown with a different icon? Are we just going to add dropdown triggers for every icon? I would move this to the app

File: libs/ui-toolkit/src/components/icon/vega-icons/vega-icon.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      'w-8 h-8': size === 8,
+      'w-10 h-10': size === 10,
+      'w-12 h-12': size === 12,
+      'w-14 h-14': size === 14,
+      'w-16 h-16': size === 16,
+    },
+    className
+  );
+  return (
+    <img
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/icon/vega-icons/vega-icon.tsx:38]: suggestion (non blocking): This is a nice tidy up, but I'd much prefer to have a regular inline svg here for easier debugging. Perhaps look into the blueprint implementation and see how they set this up so we can get inline svg

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ export const DropdownMenuContent = forwardRef<
   <DropdownMenuPrimitive.Content
     {...contentProps}
     ref={forwardedRef}
-    className="min-w-[290px] bg-neutral-200 dark:bg-white p-2 rounded z-20 dark:text-black"
+    className="min-w-[290px] bg-neutral-100 dark:bg-neutral-900 p-2 rounded z-20 text-black dark:text-white dark:border-neutral-700 border border-neutral-300"
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx:76]: chore (blocking): We should be using the vega gray scale here. The border color should be `border-vega-light-200 dark:border-vega-dark-200`

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx [hunk 4]
```diff
@@ -7,8 +7,8 @@ import { Icon } from '../icon';
 const itemClass = classNames(
   'relative flex items-center justify-between rounded-sm p-2 text-sm',
   'cursor-default hover:cursor-pointer',
-  'hover:bg-white dark:hover:bg-neutral-200',
-  'focus:bg-white dark:focus:bg-neutral-200',
+  'hover:bg-white dark:hover:bg-neutral-500',
```
Comment (mattrussell36) [inline libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx:10]: chore (blocking): Can you convert these to use the vega gray scale

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu-grid.tsx (hunk 1 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu-grid.tsx]: moved the trigger to the app in accounts container

File: libs/ui-toolkit/src/components/icon/vega-icons/vega-icon.tsx (hunk 2 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/icon/vega-icons/vega-icon.tsx]: updated to use svgs

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx (hunk 4 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx]: updated to use vega-light scale and vega-dark scale

File: libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx (hunk 3 above)
Comment (MadalinaRaicu) [reply libs/ui-toolkit/src/components/dropdown-menu/dropdown-menu.tsx]: updated to use vega-light and vega-dark scale

File: libs/accounts/src/lib/accounts-manager.tsx [hunk 5]
```diff
@@ -29,6 +31,7 @@ export const AccountManager = ({
   isReadOnly,
   pinnedAsset,
   noBottomPlaceholder,
+  gridActionsDropdownTrigger,
```
Comment (mattrussell36) [inline libs/accounts/src/lib/accounts-manager.tsx:34]: @MadalinaRaicu as discussed lets
- remove this prop
- move the dropdown trigger to libs/datagrid
- render the dropdown trigger directly where its needed

Comment (MadalinaRaicu) [reply libs/accounts/src/lib/accounts-manager.tsx]: while doing the actual changes I can see it's cleaner to:

- [ ] render the dropdown trigger directly  
- [ ] remove the prop

I think, you're right. The component is **very** small, it's not worth it to make the grid actions trigger as its own component anymore.

```html
                      <DropdownMenu
                        trigger={
                          <DropdownMenuTrigger
                            iconName="more"
                            className="hover:bg-vega-light-200 
                            dark:hover:bg-vega-dark-200 p-0.5 focus:rounded-full hover:rounded-full"
                          ></DropdownMenuTrigger>
                        }
                      >
```

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**